datastruc_ex10.py

```diff
@@ -26,11 +19,62 @@ def cyclicShift(str1: str, str2: str):
     else:
         return False
 
+# option 2
+
+
+def rotate(str: str, d: int):
+    '''
+    A function which returns a new string shifted by d length
+    '''
+    first = str[0: len(str)-d]
+    second = str[len(str)-d:]
+    return second + first
+
+
+def cyclicShift2(str1: str, str2: str):
+    '''
+    A function which recieves 2 strings and returns
+    True is str2 is a "cyclic shift" of str1 and False otherwise
+    '''
+    new_str = []
+    for i in range(len(str1)):
+        new_str.append(rotate(str1, i))
+    if str2 in new_str:
+        return True
+    else:
+        return False
+
+# b
 
-if __name__ == "__main__":
-    cyclicShift(str1="happy", str2="ppyha")
-    cyclicShift(str1="happy", str2="pypha")
-    cyclicShift(str1="YYY", str2="YYYY")
-    cyclicShift(str1="OFEK", str2="OFEK")
-    cyclicShift(str1="", str2="OFEK")
 
+def get_shifts(str1: str, str2: str):
+    '''
+    A function which returns the number of shifts
+    required to reach str2 from str1
+    '''
+    if not cyclicShift2(str1, str2):
+        return -1
+    else:
+        new_str = []
+        for i in range(len(str1)):
+            new_str.append(rotate(str1, i))
```

Comment:
Same comment here as before.. Your implementation is inefficient. DONT iterate over all the possible answers when there is a chance you already got it. We did something similar before... see ex8

Reply:
done